packages/react-components/src/components/bar-chart/barChart.tsx

```diff
@@ -34,16 +35,23 @@ export const BarChart = ({
     },
     styles,
   });
-  const { viewport } = useViewport();
+  const { viewport, setViewport, group, lastUpdatedBy } = useViewport();
   const allThresholds = [...queryThresholds, ...thresholds];
 
   const utilizedViewport = passedInViewport || viewport || DEFAULT_VIEWPORT; // explicitly passed in viewport overrides viewport group
 
+  useEffect(() => {
```

Comment:
blocking: don't useEffect here to add global listeners - react has a great built in way to handle this use case https://react.dev/learn/responding-to-events